expirationd.lua

```diff
@@ -56,106 +62,80 @@ local constants = {
 -- Task fibers
 -- ------------------------------------------------------------------------- --
 
-local function construct_key(space_id, tuple)
+-- get all fields in primary key(composite possible) from tuple
+local function construct_key(expire_index, tuple)
     return fun.map(
         function(x) return tuple[x.fieldno] end,
-        box.space[space_id].index[0].parts
+        expire_index.parts
     ):totable()
 end
 
+-- do expiration process on tuple
 local function expiration_process(task, tuple)
     task.checked_tuples_count = task.checked_tuples_count + 1
     if task.is_tuple_expired(task.args, tuple) then
         task.expired_tuples_count = task.expired_tuples_count + 1
-        task.process_expired_tuple(task.space_id, task.args, tuple)
+        task.process_expired_tuple(task.space_id, task.args, tuple, task)
```

Comment:
This will break backward compatibility. in previous versions of expirationd into a function that is not set by the user expiration_process call user defined function like `task.process_expired_tuple (task.space_id, task.args, tuple)`. It follows that the user-defined function used this data for the input parameters.